### PR TITLE
[WIP] Synchronous Filters

### PR DESCRIPTION
**Pull request by djrtwo**
### What was wrong?
[Issue 294](https://github.com/pipermerriam/web3.py/issues/294)

### How was it fixed?
- [x] Deprecate `Filter.watch` and `Filter.stop_watching`
- [x] Deprecate `LogFilter.get`
- [x] Implement `Filter.get_new_entries` and `Filter.get_all_entries`
- [x] Add deprecation warnings to docs
- [x] Add new API methods to docs

#### Cute Animal Picture

Cute animal picture

### CODE REVIEW

File: web3/utils/filters.py [hunk 1]
```diff
@@ -93,6 +94,15 @@ def _run(self):
             else:
                 sleep(self.poll_interval)
 
+    def _warn_async_deprecated(self, method_name):
+        warnings.warn(DeprecationWarning(
+            "Asyncronous filters have been deprecated "
+            "and {0} will be removed from the Filter class "
+            "in future releases.  Update your code to use work "
```
Comment (carver) [inline web3/utils/filters.py]: to ~use~ work

**Comment by djrtwo** [comment]
fixing tests and getting examples up by friday!
Sorry let this drop to the wayside for the past week

**Comment by pipermerriam** [comment]
@djrtwo looks like test failures are due to the recent change which returns all addresses in their checksummed format.  Should be simple enough to update.

**Comment by djrtwo** [comment]
@pipermerriam @carver Absolutely. I'm pretty much in a good spot for review. I can look into that tonight. What I've been putting off is the documentation/examples which I'm certain I _will not_ be able to get to before devcon.

I'll ping when I address the return address stuff tonight, and we can decide whether to merge without docs or to pass off docs to one of you.

Sorry for falling off on this. Got sucked into working on casper with some intense deadlines 😁

File: tests/core/filtering/test_contract_on_event_filtering.py [hunk 2]
```diff
@@ -11,6 +11,34 @@
...
+                                               call_as_instance):
+    if call_as_instance:
+        filter = emitter.on('LogNoArguments', {})
+    else:
+        filter = Emitter.on('LogNoArguments', {})
+
+    txn_hash = emitter.transact().logNoArgs(emitter_event_ids.LogNoArguments)
+    wait_for_transaction(web3, txn_hash)
+
+    with Timeout(30) as timeout:
```
Comment (pipermerriam) [inline tests/core/filtering/test_contract_on_event_filtering.py]: Not necessary now, but I'd very much like to rework these tests to not require anything asynchronous and to simply use the ethereum-tester backend.

File: web3/utils/filters.py [hunk 3]
```diff
@@ -106,7 +116,27 @@ def is_valid_entry(self, entry):
         """
         return True
 
+    def _filter_valid_entries(self, entries):
+        return filter(
+            lambda entry: self.is_valid_entry(entry),
```
Comment (pipermerriam) [inline web3/utils/filters.py]: I think this lambda is not necessary.  You should be able to do `return filter(self.is_valid_entry, entries)`

File: web3/utils/filters.py [hunk 4]
```diff
@@ -180,6 +210,23 @@ def get(self, only_changes=True):
...
+            "LogFilter.get has been deprecated and "
+            "will be removed from the LogFilter class in future releases. "
+            "Update your code to use the new methods: "
+            "LogFilter.get_new_entries and LogFilter.get_all_entries."
+        ))
+
+        self._ensure_not_running("get")
+
+        if only_changes:
+            log_entries = self.web3.eth.getFilterChanges(self.filter_id)
```
Comment (pipermerriam) [inline web3/utils/filters.py]: Not necessary to change, but you *could* simply delegate to `self.get_new_entries(...)` here and `self.get_all_entries(...)`

Comment (djrtwo) [reply web3/utils/filters.py]: whoops, surprised I missed that 😁

**Comment by djrtwo** [comment]
Made suggested changes, other than reworking the tests that currently require the Timeouts.

Added Issue #394 to handle documenting examples. I'm happy to pick this up post-devcon.

**Pull request merged**